unchained-commons/src/main/java/unchained/commons/contract/DefaultRegistry.java

```diff
@@ -0,0 +1,54 @@
+package unchained.commons.contract;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+class DefaultRegistry<T extends Registrable> implements Registry<T> {
+
+    private final Map<String, T> entries = new ConcurrentHashMap<>();
+
+    DefaultRegistry() {}
+
+    @Override
+    public T get(String key) {
+        return entries.get(key);
+    }
+
+    @Override
+    public boolean has(String key) {
+        return entries.containsKey(key);
+    }
+
+    @Override
+    public void register(T value) {
+        entries.put(value.registryKey(), value);
+    }
+
+    @Override
+    public Set<String> keys() {
+        return Collections.unmodifiableSet(entries.keySet());
+    }
+
+    @Override
+    public Set<T> values() {
+        final Set<T> result = new HashSet<>();
```

Comment:
Why can't we do `return Collections.unmodifiableSet(entries.values());`?

Reply:
AFAIK `entries.values()` returns `Collection`.

Reply:
(that actually doesn't work, so, how about this: `return Collections.unmodifiableSet(new HashSet<>(entries.values()));`)